### ISSUE TITLE
Not listed in monitoring during DDoS attack

### ISSUE THREAD

**Issue by daffyyyy** [sqproxy/sqproxy#129]
During an a2s attack - servers can disappear from the steam list, the website that has monitoring of the servers can't get a response from them and the network usage of the servers is high

![{79D6388F-2B44-40DA-86B0-92D422829EEE}](https://github.com/user-attachments/assets/7c95c19b-9540-42b3-9094-2fb6cd4664cc)

**Comment by spumer** [comment]
Yes, it's possible due `sqproxy` do not know who is attacker or legit user. It's try to answer to each request.
I plan change this behavior by adding whitelist to known legit ip addresses/subnets.

---

Server do not respond, but all a2s traffic redirected to sqproxy, and gameserver should works fine (if network bandwidth not utilized fully), but yes, it can dissapear from serverbrowser.

**Comment by daffyyyy** [comment]
Does sqproxy protect against this packet?
https://github.com/ValveSoftware/csgo-osx-linux/issues/1436

**Comment by daffyyyy** [comment]
I found something similar to sqproxy and that's where it redirects this packet too
https://github.com/hyperxpro/SourceEngineQueryCacher

**Comment by spumer** [comment]
> Does sqproxy protect against this packet? [ValveSoftware/csgo-osx-linux#1436](https://github.com/ValveSoftware/csgo-osx-linux/issues/1436)

Please start new issue for new questions



> I found something similar to sqproxy and that's where it redirects this packet too https://github.com/hyperxpro/SourceEngineQueryCacher

Yes, you can try it. But i recommend you replace IPtables with something like `sqredirect` - https://github.com/sqproxy/sqproxy?tab=readme-ov-file#iptables-or-any-nat-cant-help

Otherwise player connection requests can be redirected to this query cacher too and not reach gameserver.

**Comment by daffyyyy** [comment]
Is there any chance for sqproxy to accept more than 250k ips per second and still answer the query efficiently?
Currently, I have such attacks from spoofed ip and I think this may be the problem
Or if you have any solution that we can talk about I would be interested ;)

**Comment by crab-hash** [comment]
> sqproxy 是否有机会每秒接受超过 250k 个 ips 并仍然有效地回答查询？目前，我有来自欺骗性 IP 的此类攻击，我认为这可能是问题所在，或者如果您有任何我们可以讨论的解决方案，我将有兴趣;)

When the server swallows millions of packets sent by fraudsters, the server will run out of bandwidth and shut down

**Comment by daffyyyy** [comment]
Wow, maybe that's why I'm looking for a solution that can cache + handle tons of packets and limit flooding queries at the same time since it's ebpf and can be mixed with xdp i guess?

As example here  https://github.com/gnif/SteamQueryProxy
Possible to handle 900k pps+ but with c and iptables, but its nightmare to setup it for docker and multiple gameservers

**Comment by spumer** [comment]
> Wow, maybe that's why I'm looking for a solution that can cache + handle tons of packets and limit flooding queries at the same time since it's ebpf and can be mixed with xdp i guess?
> 
> As example here https://github.com/gnif/SteamQueryProxy Possible to handle 900k pps+ but with c and iptables, but its nightmare to setup it for docker and multiple gameservers

This should works fine, NFQUEUE is good choice 

And @crab-hash right, your are basically limited by your channel bandwidth

**Comment by spumer** [comment]
> Is there any chance for sqproxy to accept more than 250k ips per second and still answer the query efficiently? Currently, I have such attacks from spoofed ip and I think this may be the problem Or if you have any solution that we can talk about I would be interested ;)

I have some ideas to improve handling logic, but i can't say when they will be implemented. But this is main vector of this project. Subscribe to this repo and watch :)
